plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -484,13 +515,27 @@ public function delete($object)
       return;
     }
 
-    try
+    if ($this->batchMode)
     {
-      $this->index->getType(get_class($object))->deleteById($object->id);
+      // The document being deleted may not have been added to the index yet (if it's
+      // still queued up in $this->batchAddDocs) so create a document object representing
+      // the document to be deleted and add this document object to the batch delete
+      // queue.
+      $document = new \Elastica\Document($object->id);
+      $document->setType(get_class($object));
+
+      $this->batchDeleteDocs[] = $document;
```

Comment:
At the end of addDocument() a call to flushBatch() is made if the number of batchAddDocs is greater than batchSize. In delete(), there is no similar concept - will the queue of batchDeleteDocs only get flushed when addDocument finally triggers flushBatch()?  Should a similar condition and call to flushBatch() be included here?